modules/aws-terraform-backend/iam.tf

```
@@ -0,0 +1,112 @@
+resource "aws_iam_role" "firestartr_client" {
+  name = "firestartr-${var.tfbackend_access_role_name_suffix}"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          AWS = "arn:aws:iam::${var.aws_account_id}:root"
+        }
+      }
+    ]
+  })
+}
+
+
+resource "aws_iam_policy" "state_management" {
+  name        = "firestartr-${var.tfbackend_access_role_name_suffix}-state-access"
```

Remove firestartr- and use a full name var => 
```suggestion
  name        = "${var.tfbackend_access_role_name}-state-access"
```